src/ng/directive/select.js

```diff
@@ -288,24 +291,38 @@ var optionDirective = ['$interpolate', function($interpolate) {
             selectCtrl = parent.data(selectCtrlName) ||
               parent.parent().data(selectCtrlName); // in case we are in optgroup
 
+        function addOption(optionValue) {
+          selectCtrl.addOption(optionValue, element);
+          selectCtrl.ngModelCtrl.$render();
+          chromeHack(element);
+        }
+
         // Only update trigger option updates if this is an option within a `select`
         // that also has `ngModel` attached
         if (selectCtrl && selectCtrl.ngModelCtrl) {
 
-          if (interpolateFn) {
+          if (valueInterpolated) {
+            // The value attribute is interpolated
+            var oldVal;
+            attr.$observe('value', function valueAttributeObserveAction(newVal) {
+              if (oldVal && oldVal !== newVal) {
```

Comment:
Is it really possible to have a truthy `oldVal` that is equal to the `newVal` ?
If so, should we also refrain from adding a new option ?

Also, what happens if `oldVal` is the empty string ? Wouldn't we still want to remove the option ?